### PR TITLE
refactor: call ModelConfigAPI instead of ControllerAPI.ModelConfig

### PR DESCRIPTION
**Pull request by barrettj12**
This PR is prerequisite work to a change which will come in 4.0. While attempting to move the `Controller` facade over to model services, we realised that the controller facade *should* be a multi-model facade. Currently it is a model-scoped facade. However, we can't change it over right now due to the `ModelConfig` facade method, which requires the facade to be scoped to a single model.

So in this PR, we replace all usages of `Controller.ModelConfig` with the `ModelConfig` facade. This ensures that 3.6 clients will continue to be compatible with 4.0 controllers when we change the `Controller` facade in 4.0.

This PR does not include any facade changes, but we have removed the `ModelConfig` method from the controller API client to discourage people from using that facade method. Furthermore, we have added a deprecation warning to the facade method. In 4.0, we will remove this facade method and bump the facade version.

## Checklist

- [x] Code style: imports ordered, good names, simple structure, etc
- ~[ ] Comments saying why design decisions were made~
- [x] Go unit tests, with comments saying what you're testing
- ~[ ] [Integration tests](https://github.com/juju/juju/tree/main/tests), with comments saying what you're testing~
- ~[ ] [doc.go](https://discourse.charmhub.io/t/readme-in-packages/451) added or updated in changed packages~

## QA steps

Affected commands are `upgrade-model`, `destroy-controller`, `kill-controller`, `show-controller`. The following should test all of the code paths that have been modified.

Bootstrap Juju, run `show-controller`, and check that the controller model version displays correctly:
```
$ juju show-controller | grep controller-model-version
    controller-model-version: 3.6-beta2.1
```

Run `upgrade-model` on the controller model, and check that it raises an error:
```
$ juju switch controller
$ juju upgrade-model
ERROR use upgrade-controller to upgrade the controller model
```

Run `destroy-controller` and check that the controller is successfully destroyed.

Bootstrap Juju again, run `kill-controller` and check that the controller is successfully destroyed.

## Links

**Jira card:** JUJU-6350

### CODE REVIEW

File: cmd/juju/commands/upgradecontroller.go [hunk 1]
```diff
@@ -96,12 +93,19 @@ type upgradeControllerCommand struct {
 
 // ControllerAPI defines the controller API methods.
 type ControllerAPI interface {
-	CloudSpec(modelTag names.ModelTag) (environscloudspec.CloudSpec, error)
-	ControllerConfig() (controller.Config, error)
 	ModelConfig() (map[string]interface{}, error)
 	Close() error
 }
```
Comment (SimonRichardson) [inline cmd/juju/commands/upgradecontroller.go:98]: Rename this or remove it.

**Review by tlm — APPROVED** [review]
lgtm, I have run all the tests I can think of and this looks good.

Next step will be to cut a new facade version in 3.6 that removes this method and then we can nurse that all the way up into 4.0.

I have checked the Terraform provider wasn't using this. Python may use this so I think cut a new version in 3.6 removing the command and that covers our bases. We can then just support the new version only in 4.0.

File: cmd/juju/commands/upgradecontroller.go [hunk 2]
```diff
@@ -356,12 +333,6 @@ func (c *upgradeControllerCommand) upgradeController(
 	}
 	defer modelUpgrader.Close()
 
-	controllerClient, err := c.getControllerAPI()
```
[the remark this answers is not in the document]
Comment (barrettj12) [reply cmd/juju/commands/upgradecontroller.go:359]: yeah, probably was being used at one point but someone forgot to remove it

File: cmd/juju/controller/kill.go [hunk 3]
```diff
@@ -116,8 +116,14 @@ func (c *killCommand) Run(ctx *cmd.Context) error {
 		ctx.Infof("Unable to open API: %s\n", err)
 	}
 
+	controllerModelConfigAPI, err := c.getControllerModelConfigAPI()
+	if err != nil {
+		return fmt.Errorf("cannot connect to model config API: %w", err)
+	}
+	defer func() { _ = controllerModelConfigAPI.Close() }()
```
[the remark this answers is not in the document]
Comment (barrettj12) [reply cmd/juju/controller/kill.go:123]: It's already being done everywhere I can see

**Pull request merged**